crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs

```diff
@@ -772,9 +1060,12 @@ pub struct DeutschebankReversalRequest {
 
 impl TryFrom<&PaymentsCancelRouterData> for DeutschebankReversalRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(_item: &PaymentsCancelRouterData) -> Result<Self, Self::Error> {
+    fn try_from(item: &PaymentsCancelRouterData) -> Result<Self, Self::Error> {
         Ok(Self {
-            kind: DeutschebankTransactionKind::Directdebit,
+            kind: match item.is_three_ds() && matches!(item.payment_method, PaymentMethod::Card) {
+                true => DeutschebankTransactionKind::Creditcard3ds20,
+                false => DeutschebankTransactionKind::Directdebit,
+            },
```

Comment:
Add explicit matching where
if sepa then value should be Directdebit
if cards + 3ds the value should be Creditcard3ds20
else not supported error